front_end/content.py

```diff
@@ -292,6 +293,7 @@ def get_problems(self, course_id, assignment_id, show_hidden=True):
     def get_assignment_statuses(self, course_id, user_id):
         sql = '''SELECT assignment_id,
                         title,
+                        SUM(passed) AS num_passed,
```

Comment:
Let's put the "in progress" logic in SQL rather than in Python. You should be able to add something like the following to your outer SELECT statement:

SUM(passed) > 0 AND SUM(passed) < COUNT(assignment_id) AS in_progress

This should give you a 1 if it's in progress and a 0 if it hasn't been started.